### ISSUE TITLE
Sub-components as `<Component.SubComponent>` trig eslint errors in MDX files

### ISSUE THREAD

**Issue by vripoche** [mdx-js/eslint-mdx#445]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.x.x

### Steps to reproduce

Create a MDX file with imported Component that have sub-components as:

`doc.mdx`:
```
import { DefinitionList } from `./DefinitionList`

<DefinitionList>
   <DefinitionList.Title>Hello World !</DefinitionList.Title>
<DefinitionList>
```
And `DefinitionList` component:
`DefinitionList.js`:
```
import React from 'react'

export const DefinitionList = ({ children }) => <dl>{children}</dl>
DefinitionList.Title =  ({ children }) => <dt>{children}</dt>
```

### Expected behavior

Should not be any eslint errors

### Actual behavior

On MDX linting (`eslint ./doc.mdx`), we have this error: `error  'DefinitionList.Title' is not defined        react/jsx-no-undef`.

Note: no errors were triggered on last `1.17.1` version.

### Runtime

Node v17

### Package manager

npm v6

### OS

macOS

### Build and bundle tools

webpack

**Comment by oliviadawd** [comment]
im getting this error as well!
